stdlib/public/core/DoubleWidth.swift.gyb

```diff
@@ -455,12 +446,21 @@ extension DoubleWidth : FixedWidthInteger {
   public func dividingFullWidth(
     _ dividend: (high: DoubleWidth, low: DoubleWidth.Magnitude)
   ) -> (quotient: DoubleWidth, remainder: DoubleWidth) {
-    let lhs = DoubleWidth<DoubleWidth<Base>>(dividend)
-    let rhs = DoubleWidth<DoubleWidth<Base>>(self)
-    let (quotient, remainder) = lhs.quotientAndRemainder(dividingBy: rhs)
-
-    // FIXME(integers): check for overflow of quotient and remainder
-    return (DoubleWidth(quotient.low), DoubleWidth(remainder.low))
+    let other = DoubleWidth<DoubleWidth>(dividend)
+    let (quotient, remainder) =
+      Magnitude._divide(other.magnitude, by: self.magnitude)
+    guard DoubleWidth.isSigned else {
+      return (DoubleWidth(quotient), DoubleWidth(remainder))
+    }
+    let quotient_ = (self.high < (0 as High)) != (other.high.high < (0 as High))
```

Comment:
This part looks sufficiently similar to the one above (inside `quotientAndRemainder`. I wonder if it can be extracted into a separate `@_inline(always)` function without affecting performance.

Reply:
Unfortunately, not possible because they’re dealing with different types (this is a heterogeneous operation).